sys/net/network_layer/ng_sixlowpan/iphc/ng_sixlowpan_iphc.c

```diff
@@ -509,6 +513,16 @@ bool ng_sixlowpan_iphc_encode(ng_pktsnip_t *pkt)
                                   &iid, sizeof(eui64_t)) >= 0) {
                     addr_comp = true;
                 }
+
+                if ((!try_long) || (ng_netapi_get(netif_hdr->if_pid,
+                                                  NETCONF_OPT_ADDRESS_LONG, 0,
+                                                  &l2src, sizeof(l2src)) < 0)) {
+                    if (ng_netapi_get(netif_hdr->if_pid, NETCONF_OPT_ADDRESS,
```

Comment:
I just reversed the merger at a different spot (#3396). I'd be more inclined to do it more similar as I did it there